client/v2/internal/util/util.go

```diff
@@ -24,3 +36,56 @@ func ResolveMessageType(resolver protoregistry.MessageTypeResolver, descriptor p
 
 	return dynamicpb.NewMessageType(descriptor)
 }
+
+// IsSupportedVersion is used to determine in which version of a module / sdk a rpc was introduced.
+// It returns false if the rpc has comment for an higher version than the current one.
+func IsSupportedVersion(input string) bool {
+	return isSupportedVersion(input, buildInfo)
+}
+
+// isSupportedVersion is used to determine in which version of a module / sdk a rpc was introduced.
+// It returns false if the rpc has comment for an higher version than the current one.
+// It takes a buildInfo as argument to be able to test it.
+func isSupportedVersion(input string, buildInfo *debug.BuildInfo) bool {
+	if input == "" || buildInfo == nil {
+		return true
+	}
+
+	moduleName, version := parseSinceComment(input)
+	for _, dep := range buildInfo.Deps {
+		if !strings.Contains(dep.Path, moduleName) {
+			continue
+		}
+
+		if version <= dep.Version {
+			return true
+		}
+
+		return false
+	}
+
+	return true // if cannot find the module consider it's supported
+}
+
+// parseSinceComment parses the `// Since: cosmos-sdk v0.xx` comment on rpc.
+func parseSinceComment(input string) (string, string) {
+	var (
+		moduleName string
+		version    string
+	)
+
+	input = strings.ToLower(input)
+	input = strings.ReplaceAll(input, "cosmos sdk", "cosmos-sdk")
+
+	re := regexp.MustCompile(`\/\/\s?since: (\S+) (\S+)`)
```

Comment:
The comment above says "Since", yet this regex uses "since", but we aren't using the case insensitive flag "(?i)" We could simply use "[sS]ince". I highly recommend removing the strings.ToLower then simply using that regexp suggestion that I've made. This way if some module's name is for example PbValidator, it won't be returned as "pbvalidator"

Reply:
Case insensitive definitely makes sense, but I'll still need to return the module name in lowercase.
I'll move down the strings.ToLower then